### PR TITLE
cptofs: Preserve timestamps

### LINKED ISSUE #393: Preserving file times when copying with cptofs
I need all my files inside my image to have a UNIX timestamp of 1. If I just use `cptofs` normally, it uses the current time. I can use `libfaketime` to fool `cptofs` into thinking that the current time is 1, but oddly enough all the timestamps in the image come out as 2 instead of 1.

So now I'm confused what's going on. Can `cptofs` get an option to preserve file times, or if not, does anyone know why all my timestamps are getting incremented by 1 before making it into the image?

If I tell `cptofs` that the time is 0, all my timestamps in the image are 1, so I'm not stuck, but it would be nice if this behaved more intuitively.

cc @dezgeg

### PR DESCRIPTION
**Pull request by dezgeg**
Preserves mtime and atime of inodes copied in/out of filesystems. I've made it unconditional here, let me know if that should be behind some command line flag instead.

Fixes #393

cc @copumpkin

---
This change is [Reviewable](https://reviewable.io/reviews/lkl/linux/394)

### CODE REVIEW

**Comment by tavip** [comment]
@lkl-jenkins: test this please

File: tools/lkl/cptofs.c [hunk 1]
```diff
@@ -390,6 +409,19 @@ static int do_entry(const char *_src, const char *_dst, const char *name)
 		printf("skipping %s: unsupported entry type %d\n", src, type);
 	}
 
+	if (!ret) {
+		if (cptofs) {
+			struct lkl_timespec lkl_ts[] = { atime, mtime };
```
Comment (tavip) [inline tools/lkl/cptofs.c]: Could you please add a new line here to keep checkpath happy?

File: tools/lkl/cptofs.c [hunk 2]
```diff
@@ -390,6 +409,19 @@ static int do_entry(const char *_src, const char *_dst, const char *name)
...
+			ret = lkl_sys_utimensat(-1, dst, lkl_ts, LKL_AT_SYMLINK_NOFOLLOW);
+		} else {
+			struct timespec ts[] = {
+				{ .tv_sec = atime.tv_sec, .tv_nsec = atime.tv_nsec, },
+				{ .tv_sec = mtime.tv_sec, .tv_nsec = mtime.tv_nsec, },
+			};
```
Comment (tavip) [inline tools/lkl/cptofs.c]: and here as well

File: tools/lkl/cptofs.c (hunk 1 above)
Comment (dezgeg) [reply tools/lkl/cptofs.c]: Amended.

**Pull request merged**